cmd/faucet/faucet.go

```diff
@@ -745,17 +761,30 @@ func (f *faucet) loop() {
 			f.lock.RUnlock()
 		}
 	}()
+
 	// Wait for various events and assing to the appropriate background threads
+	subscribeTimeoutTimer := time.NewTimer(subscribeTimeout)
+
 	for {
 		select {
 		case head := <-heads:
+			if !subscribeTimeoutTimer.Stop() {
```

Comment:
No need to update timer here, all goes to the `update` chan below.

Reply:
no, you need check `Stop()` result